paprika/setup.py

```diff
@@ -31,732 +26,685 @@ def _get_installed_benchmarks():
-
-        if build:
-            # Here, we build desolvated windows and pass the files to the OpenFF Evaluator.
-            # These files are stored in `self.desolvated_window_paths`.
-            self.build_desolvated_windows(guest_orientation)
-            if generate_gaff_files:
-                generate_gaff(mol2_file=self.benchmark_path.joinpath(self.host_yaml["structure"]),
-                              residue_name=self.host_yaml["resname"],
-                              output_name=self.host,
-                              directory_path=self.directory,
-                              gaff=gaff_version)
-                if guest:
-                    generate_gaff(mol2_file=self.benchmark_path.joinpath(
-                        self.guest).joinpath(self.guest_yaml["structure"]),
-                                  output_name=self.guest,
-                                  residue_name=self.guest_yaml["name"],
-                                  directory_path=self.directory,
-                                  gaff=gaff_version)
-        if not build:
-            self.populate_window_list(input_pdb=os.path.join(self.directory, f"{self.host}-{self.guest}.pdb" if self.guest is not None
-            else f"{self.host}.pdb"))
-
-    def parse_yaml(self, installed_benchmarks, guest_orientation):
+        # Extract the host from the full structure.
+        if not host_atom_indices:
+            host_atom_indices = range(len(structure.atoms))
+
+        host_structure = structure[
+            "@" + ",".join(map(lambda x: str(x + 1), host_atom_indices))
+        ]
+
+        # noinspection PyTypeChecker
+        center_of_mass: np.ndarray = pmd.geometry.center_of_mass(
+            host_structure.coordinates, masses=np.ones(len(host_structure.coordinates))
+        )
+
+        # Remove the COM from the host coordinates to make alignment easier.
+        structure.coordinates -= center_of_mass
+        host_structure.coordinates -= center_of_mass
+
+        # Find the principal components of the host, take the two largest, and find
+        # the vector orthogonal to that. Use that vector to align with the z-axis.
+        # This may not generalize to non-radially-symmetric host molecules.
+        inertia_tensor = np.dot(
+            host_structure.coordinates.transpose(), host_structure.coordinates
+        )
+
+        eigenvalues, eigenvectors = np.linalg.eig(inertia_tensor)
+        order = np.argsort(eigenvalues)
+
+        _, axis_2, axis_1 = eigenvectors[:, order].transpose()
+
+        cavity_axis = np.cross(axis_1, axis_2)
+
+        # Add dummy atoms which will be used to align the structure.
+        cls.add_dummy_atoms_to_structure(structure, [np.array([0, 0, 0]), cavity_axis])
+
+        # Give atoms uniform mass so that the align code uses the center
+        # of geometry rather than the center of mass.
+        for atom in structure.atoms:
+            atom.mass = 1.0
```

Comment:
Do we have to worry about undoing this to further process the `structure`, since it is returned from this function?

Reply:
We do! At the end of the function we reload the original structure from disk, set the coordinates on that, and then return that so that the structure remains the same and only the coordinates change!

Reply:
Ah, perfect. Clever.